duckduckgo/Android
# Remote Messaging - Home tab integration (4)

Task/Issue URL: https://app.asana.com/0/72649045549333/1199214002677571/f

### Description
This PR introduces the UI component to show remote messages. It also brings:
- logic to show a remote message on the home tab
- priorities when message competes with other CTAs or the onboarding
- pixel logic

### Steps to test this PR

_No message during onboarding_
- Fresh install (use internal build)
- Do not select default browser
- Open the app and go through the onboarding screens
- Ensure you don't see any remote message on the home tab if there's a Dax dialog

_Remote message has priority over widget/survey CTA_
- after completing the onboarding, ensure first remote message appears
- ensure pixel `m_remote_message_shown` and `m_remote_message_shown_unique`
- Ensure Widget CTA doesn't appear if message present
- visit any site, and open a new tab
- Ensure message appears on the new home tab 
- ensure pixel `m_remote_message_shown`(but not the unique one)

._New message_
- Now click on the "X" to dismiss the remote message
- ensure pixel `m_remote_message_dismissed` sent
- Ensure Widget CTA appears
- Click on the Fire button
- Ensure a new message appears
- Click on "Get tips"
- Ensure it performs a search of "covid 19"
- Ensure "m_remote_message_primary_action_clicked" pixel sent
- Open a new tab
- Ensure remote message is gone

._Message with favorites and AppTP banner_
- Add 2 sites as favorites
- Click on the fire button
- Ensure new message "default browser" appears
- Now enable AppTP in Settings
- Go back to the home tab
- Ensure AppTP banner, remote message and favorites are visible
- Click on "Later"
- Ensure "m_remote_message_secondary_action_clicked" pixel sent


### UI changes
| New |
| ------ |
| <img src ="https://user-images.githubusercontent.com/4747865/154048312-82828ace-e001-4b02-aa14-91ce109fb77a.png" width="360px"/> |
| <img src ="https://user-images.githubusercontent.com/4747865/154048618-8883a45b-bee2-4761-9ad1-cba50f821a6e.png" width="360px"/> |
| <img src ="https://user-images.githubusercontent.com/4747865/154048734-b33e30d0-8d66-46aa-b1df-38dd2a334624.png" width="360px"/> |
